stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/RegistrationPageResource.java

```diff
@@ -89,7 +102,29 @@ public Response get(
         }
 
         Idp idp = idpStubsRepository.getIdpWithFriendlyId(idpName);
-        return Response.ok(new RegistrationPageView(idp.getDisplayName(), idp.getFriendlyId(), errorMessage.orElse(NO_ERROR).getMessage(), idp.getAssetId())).build();
+        return Response.ok(new RegistrationPageView(idp.getDisplayName(), idp.getFriendlyId(), errorMessage.orElse(NO_ERROR).getMessage(), idp.getAssetId(), null)).build();
+    }
+
+    @GET
+    @Path(Urls.PRE_REGISTER_PATH)
+    public Response get(
+            @PathParam(Urls.IDP_ID_PARAM) @NotNull String idpName,
+            @QueryParam(Urls.ERROR_MESSAGE_PARAM) Optional<ErrorMessageType> errorMessage) {
+
+        Idp idp = idpStubsRepository.getIdpWithFriendlyId(idpName);
+        IdpSession idpSession = new IdpSession(
+                new SessionId(UUID.randomUUID().toString()),
+                null,
+                null,
+                null,
+                null,
+                null,
+                null,
+                Optional.of(UUID.randomUUID()));
+        final SessionId sessionId = idpSessionRepository.createSession(idpSession);
+        return Response.ok(new RegistrationPageView(idp.getDisplayName(), idp.getFriendlyId(), errorMessage.orElse(NO_ERROR).getMessage(), idp.getAssetId(), null))
```

Comment:
Shouldn't the final parameter here be `Urls.PRE_REGISTER_PATH` rather than `null`?

Reply:
Yes, that's why the tabs were still showing. I've updated accordingly.